### PR TITLE
Avatar Picker: Introduce horizontal grid layout

### LINKED ISSUE #345: Quick Editor: Horizontally scrolling avatars
https://www.figma.com/design/xgXj4jig9m0DYfKjFq6swG/QE-Native-Experience?node-id=44-1094&t=6N8vBEajkfr2yJEe-0

### PR DESCRIPTION
**Pull request by pinarol**
Closes #345 

From designs:

Screenshot 2024-08-15 at 15 34 50

### Description

Introduces the horizontally scrolling avatars. For now i didn't implement anything interactive like using number of avatars or height of the modal to decide the layout type. 

https://github.com/user-attachments/assets/b9476158-388e-4b33-a8d7-31709469e66e

- Extracts the avatar view to a separate View named `AvatarPickerAvatarView` to be able to use it both from horizontal & vertical grids.
- Updates the UI according to designs to put the content inside the borders whether it's vertical or horizontal. 
- Extracted the border code to an internal modifier function because it is used by error views too.
- Updated the demo app to add the layout options.

### Testing Steps

- SwiftUI demo app > Avatar picker view > Play around with the "content layout" parameter.

### CODE REVIEW

File: Sources/GravatarUI/SwiftUI/AvatarPicker/Views/AvatarPickerAvatarView.swift [hunk 1]
```diff
@@ -0,0 +1,59 @@
+import SwiftUI
+
+struct AvatarPickerAvatarView: View {
+    let avatar: AvatarImageModel
+    let shouldSelect: () -> Bool
+    let onAvatarTap: (AvatarImageModel) -> Void
+    let onImageSelected: (UIImage) -> Void
```
Comment (etoledom) [inline Sources/GravatarUI/SwiftUI/AvatarPicker/Views/AvatarPickerAvatarView.swift]: I don't think we need `onImageSelected` here.

This is the callback when a new image is selected from the + cell in the grid, which only exists on the Vertical layout.

Maybe the name is confusing? 🤔

Comment (etoledom) [reply Sources/GravatarUI/SwiftUI/AvatarPicker/Views/AvatarPickerAvatarView.swift]: Also `HorizontalAvatarGrid` shouldn't need this.

**Review by etoledom — COMMENTED** [review]
I see some flashing when selecting a different cell on Horizontal layout:

CleanShot 2024-08-16 at 11 29 36

I don't see this happening on Vertical layout.

@pinarol - Just wondering if you have seen this also

**Comment by etoledom** [comment]
One more thing:

On the Horizontal layout, it doesn't seem obvious that I can scroll horizontally when I see 3 avatars aligned on the screen. (Testing on iPhone 15 sim)

The design shows a bit of a 4th avatar appearing from the right. This helps the user see that there's more content and discover the scrolling. 

Can we do this also and have this extra bit of 4th avatar appearing independent of the screen width?
I know that on UICollectionView we can have a layout based on percentages of the screen, maybe there's a similar thing on SwiftUI 🤔

File: Sources/GravatarUI/SwiftUI/AvatarPicker/Views/AvatarPickerAvatarView.swift (hunk 1 above)
Comment (pinarol) [reply Sources/GravatarUI/SwiftUI/AvatarPicker/Views/AvatarPickerAvatarView.swift]: I see, yes the name is confusing then. I'll rename and remove from here.

**Comment by pinarol** [comment]
I made some changes to avoid glitches. The horizontal version uses fixed size avatars now.

Also, I tried calculating the avatar size with the help of GeometryReader to display 3.25 avatars on every device but it is messing up the height of `HorizontalAvatarGrid`. I tried setting the height manually but no good way of doing it. So I left it. I made the avatar larger and the space between each avatar smaller. This way it is also more close to the designs. I think this will make it less likely to happen. 

Also, it is not great to always show 3.25 avatars, it is causing giant avatars in landscape mode and in ipad. This needs a more sophisticated formula. But the bigger problem is the height of `HorizontalAvatarGrid`.  
Can you try again? @etoledom

**Review by etoledom — APPROVED** [review]
Looking great!
Thanks for the changes 🙏

**Pull request merged**